ttnn/cpp/ttnn/operations/creation.hpp

```diff
@@ -173,11 +171,36 @@ struct FullLikeWith {
 
 struct ZerosLike : FullLikeWith<0.0f> {};
 struct OnesLike : FullLikeWith<1.0f> {};
-struct EmptyLike : FullLikeWith<0.0f> {};
 
 inline constexpr ZerosLike zeros_like{};
 inline constexpr OnesLike ones_like{};
-inline constexpr EmptyLike empty_like{};
+
+struct Empty {
+   static ttnn::Tensor invoke(
+    const ttnn::Shape& shape,
+    const DataType& dtype,
+    const Layout& layout,
+    Device* device,
+    const std::optional<MemoryConfig>& memory_config = ttnn::DRAM_MEMORY_CONFIG) {
+        TT_FATAL(memory_config.has_value(), "MemoryConfig cannot be null");
+        return create_device_tensor(shape, dtype, layout, device, *memory_config);
+    }
+};
+
+struct EmptyLike {
+   static ttnn::Tensor invoke(
+    const ttnn::Tensor& tensor,
+    const std::optional<DataType>& dtype = std::nullopt,
+    const std::optional<Layout>& layout = std::nullopt,
+    const std::optional<std::reference_wrapper<Device>>& device_arg = std::nullopt,
+    const std::optional<MemoryConfig>& memory_config = std::nullopt) {
+    Device* device = device_arg.has_value() ? &(device_arg.value().get()) : tensor.device();
```

Comment:
Here the std::optional makes perfect sense because we can get the value from the tensor.